Libplanet/Blockchain/BlockChain.cs

```diff
@@ -1434,9 +1453,55 @@ public IImmutableSet<TxId> GetStagedTransactionIds()
             double evalDuration = (DateTimeOffset.Now - evaluateActionStarted).TotalMilliseconds;
             _logger.Debug(evalEndMsg, block.Index, block.Hash, evalDuration);
 
+            // Prepare TxExecutions
+            var txExecutions = new List<TxExecution>();
+            IEnumerable<IGrouping<TxId, ActionEvaluation>> evaluationsPerTxs = evaluations
+                .Where(e => e.InputContext.TxId is { })
+                .GroupBy(e => e.InputContext.TxId.Value);
+            foreach (IGrouping<TxId, ActionEvaluation> txEvals in evaluationsPerTxs)
+            {
+                TxId txid = txEvals.Key;
+                IAccountStateDelta prevStates = txEvals.First().InputContext.PreviousStates;
+                ActionEvaluation evalSum = txEvals.Last();
+                TxExecution txExecution;
+                if (evalSum.Exception is { } e)
+                {
+                    txExecution = new TxFailure(block.Hash, txid, e.InnerException ?? e);
```

Comment:
It seems that care `.InnerException` only. is it an intended trade-off or any other reason?

Reply:
It's merely for suppressing the compiler's null check, because `Exception.InnerException` is `Exception?` API-wise.